tests/test_video_capturer.py

```diff
@@ -0,0 +1,70 @@
+import time
+import numpy as np
+import mock
+import random
+from mock import patch
+
+from src.jetson.video_capturer import VideoCapturer
+def mock_read():
+    img = np.zeros((300, 300, 3), np.uint8)
+    img[:] = (random.randint(0,255), random.randint(0,255), random.randint(0,255))
+    return None, img
+
+@mock.patch('cv2.VideoCapture')
+class TestCapturer():
+    '''
+    Tests in this class are for the VideoCapturer class found in src/video_capturer.py
+    '''
+
+    def test_init(self, mock_capture):
+        mock_capture.return_value.read.return_value = mock_read() 
+        self.capturer = VideoCapturer(False)
+        '''
+        Tests constructor
+        Checks:
+            - Update thread is created
+            - Capturer is not null (interface with camera established)
+            - First captured frame is not null
+        '''
+        assert self.capturer.t1.isAlive()
+        assert self.capturer.capture is not None
+        assert self.capturer.frame is not None
+
+    def test_update(self, mock_capture):
+        mock_capture.return_value.read.return_value = mock_read()
+        self.capturer = VideoCapturer(False)
+        '''
+        Tests 'update' function
+        Checks:
+            - Frames captured are not null
+            - Every frame is different from the previous frame
+        '''
+        assert self.capturer.frame is not None
+        oldFrame = self.capturer.frame.copy()
+        time.sleep(.02) # sleep long enough to ensure thread t1 updates
+        assert oldFrame is not self.capturer.frame
```

Comment:
I think this is used incorrectly see [is vs is not](https://realpython.com/python-is-identity-vs-equality/). essentially is not compares the memory address when we should be comparing the values instead by using `!=` which will invoke the `__eq__` method of the left-hand object.

Reply:
I couldn't figure out how to compare the contents of a numpy array. I think the use of channels makes the memory structure ambiguous. But I guess if I'm only comparing the frame address I don't really need to randomize the contents of the frames.

Reply:
That sounds fine to me. Also check out [this](https://www.kite.com/python/answers/how-to-compare-two-numpy-arrays-in-python#:~:text=Use%20np.,two%20NumPy%20arrays%20are%20equivalent.) link for comparing numpy arrays.

Reply:
It looks like this solution is working for both random generated frames and frames captured from a real camera. Thanks!